jsontokenizer_test.go

```diff
@@ -0,0 +1,214 @@
+package gojsonsm
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+)
+
+func testTokenizedStep(t *testing.T, tok *jsonTokenizer, expectedToken tokenType, expectedTokenStr string) {
+	t.Helper()
+
+	token, tokenData, err := tok.Step()
+	if err != nil {
+		t.Fatalf("encountered stepping error: %s", err)
+	}
+
+	if token != expectedToken {
+		t.Fatalf("Expected token `%s` but got `%s`", tokenToText(expectedToken), tokenToText(token))
+	}
+
+	if bytes.Compare(tokenData, []byte(expectedTokenStr)) != 0 {
+		t.Fatalf("Expected token data `%s` but got `%s`", expectedTokenStr, tokenData)
+	}
+}
+
+func testTokenizedValue(t *testing.T, dataBytes []byte, token tokenType, tokenData string) {
+	t.Helper()
+
+	// Set up the tokenizer
+	var tok jsonTokenizer
+	tok.Reset(dataBytes)
+
+	// Check that we receive the correct token
+	testTokenizedStep(t, &tok, token, tokenData)
+
+	// Check that we get end after the token
+	//testTokenizedStep(t, &tok, tknEnd, "")
```

Comment:
Why did you left commented out test?

Reply:
Nice catch...  I commented it out because it wasn't working correctly...  Will fix!